core/src/main/java/com/microsoft/applicationinsights/common/SanitizationUtils.java

```diff
@@ -0,0 +1,54 @@
+package com.microsoft.applicationinsights.common;
+
+import java.text.StringCharacterIterator;
+
+/**
+ * Created by dhdoshi on 09/10/2017
+ *
+ * This class provides utility functions to sanitize strings
+ * for various formats. Currently it supports JSON sanitization
+ */
+public final class SanitizationUtils {
+
+    /**
+     * This method appends escape characters to input String to prevent
+     * JSON Sanitization failure
+     * @param text
+     * @return Sanitized String suitable for JSON
+     */
+    public static String sanitizeStringForJSON(String text) {
+
+        final StringBuilder result = new StringBuilder();
+        StringCharacterIterator iterator = new StringCharacterIterator(text);
+        for (char curr = iterator.current(); curr != iterator.DONE; curr = iterator.next()) {
+            if( curr == '\"' ){
```

Comment:
Do we need to take care of the escaped single quote as well? `\'`

Reply:
@Dmitry-Matveev added \' escape character and also added the test case.